packages/now-build-utils/src/detectors/index.ts

```diff
@@ -0,0 +1,51 @@
+import AggregateError from 'aggregate-error';
+import DetectorFilesystem from './filesystem';
+import { Detector, DetectorParameters, DetectorResult } from '../types';
+
+import cra from './create-react-app';
+import docusaurus from './docusaurus';
+import eleventy from './eleventy';
+import ember from './ember';
+import hexo from './hexo';
+import hugo from './hugo';
+import gatsby from './gatsby';
+import preact from './preact';
+
+export { DetectorFilesystem };
+
+export const detectors: Detector[] = [
+  cra,
+  docusaurus,
+  eleventy,
+  ember,
+  gatsby,
+  hexo,
+  hugo,
+  preact,
+];
+
+function firstTruthy<T>(promises: Promise<T>[]) {
+  return new Promise<T>((resolve, reject) => {
+    const errors: Array<Error> = [];
+    let unresolved = promises.length;
+    for (const p of promises) {
+      p.then(v => {
+        if (v || --unresolved === 0) {
+          resolve(v);
```

Comment:
Why does it resolve even when there are no truthy values? I would expect it to reject if none are truthy. Or maybe return `null`?

Reply:
If it's the final promise (unresolved === 0) then it needs to resolve _something_, so it resolves to the _falsy_ value of the final promise. `null` would work too for consistency, but the point is that it's already being resolved to an falsy value.